tactics/eqdecide.ml

```diff
@@ -156,8 +156,9 @@ open Proofview.Notations
 (* spiwack: a PatternMatchingFailure wrapper around [Hipattern]. *)
 
 let match_eqdec sigma c =
+  Proofview.tclENV >>= fun env ->
```

Comment:
Pass the environment functionally instead. All of the calls are in the context of a goal.